### PR TITLE
feat:dark mode implementation

### PR DESCRIPTION
**Pull request by suyash5641**
Implemented dark mode in astro

https://github.com/user-attachments/assets/d8972d24-b1ff-42e8-8642-8f9ed5aac44a

### CODE REVIEW

**Comment by suyash5641** [comment]
@yaralahruthik  I have made the changes for dark mode change . Waiting for feedback

File: src/global.css
Comment (yaralahruthik) [inline src/global.css:1]: @suyash5641 can we further simplify the approach to setting the colors?

I was looking at something like how ShadCN does it for Tailwind: https://ui.shadcn.com/docs/installation/manual

Comment (suyash5641) [reply src/global.css:1]: Yes I looked at this when I browsing through the document but missed to implement this . Will certainly do this

File: src/global.css [hunk 1]
```diff
@@ -1,10 +1,50 @@
...
   scrollbar-gutter: stable;
 }
+
+.lightThemeIcon {
+  fill: var(--background-color-dark);
+}
+
+.darkThemeIcon {
+  fill: transparent;
+}
```
Comment (yaralahruthik) [inline src/global.css:21]: The styles for Icon could live in the `ThemeIcon.astro` file

File: src/components/ThemeIcon.astro [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      const element = document.documentElement;
+      element.classList.toggle("dark");
+      const isDark = element.classList.contains("dark");
+      localStorage.setItem("theme", isDark ? "dark" : "light");
+    };
+    toggleBtn?.addEventListener("click", handleToggleClick);
+  };
+  setToggleListener();
+  document.addEventListener("astro:after-swap", setToggleListener);
+</script>
```
Comment (yaralahruthik) [inline src/components/ThemeIcon.astro:28]: Is this the convention in Astro? I thought we put the JS logic inside the code fences (`---`)

File: src/global.css (hunk 1 above)
Comment (suyash5641) [reply src/global.css:21]: Ok under the style section right ??

**Review by yaralahruthik — CHANGES_REQUESTED** [review]

File: src/global.css (hunk 1 above)
Comment (yaralahruthik) [reply src/global.css:21]: Yes!

File: src/components/ThemeIcon.astro (hunk 2 above)
Comment (suyash5641) [reply src/components/ThemeIcon.astro:28]: I am also not sure in the docs I didn't found any fences (---) 
https://docs.astro.build/en/tutorial/6-islands/2/ 
May be I have missed ??

Comment (suyash5641) [reply src/components/ThemeIcon.astro:28]: Ok I will check and if required make the necessary changes

**Comment by suyash5641** [comment]
@yaralahruthik I have made the required changes , please have a look at the changes 
Thanks

**Comment by suyash5641** [comment]
@yaralahruthik I think there was no need of adding these packages 
tailwindcss-animate class-variance-authority
In this doc  https://ui.shadcn.com/docs/installation/manual I saw that packages were added , but as per current change I don't think we need these packages as of now

**Comment by suyash5641** [comment]
To disable lint warning Unknown at rule @tailwindcss(unknownAtRules)
@tailwind base;
@tailwind components;
@tailwind utilities;

change this setting in vs code 

image

**Comment by yaralahruthik** [comment]
> @yaralahruthik I think there was no need of adding these packages tailwindcss-animate class-variance-authority In this doc https://ui.shadcn.com/docs/installation/manual I saw that packages were added , but as per current change I don't think we need these packages as of now . So should I revert the changes in package.json and pnpm-lock.yaml files ??

I agree.

File: src/global.css [hunk 3]
```diff
@@ -1,10 +1,30 @@
-:root {
-  --color-primary: 0deg 0% 0%;
-  --color-secondary: 0deg 0% 24%;
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
```
Comment (yaralahruthik) [inline src/global.css:3]: You should not be doing this I believe. This results in duplicate classes. Could you check this issue? I remember the duplication issue. The way we add Tailwind in Astro is a bit different.

**Review by yaralahruthik — CHANGES_REQUESTED** [review]

File: src/global.css (hunk 3 above)
Comment (suyash5641) [reply src/global.css:3]: Ok I will check this

Comment (yaralahruthik) [reply src/global.css:3]: @suyash5641 in fact check this in the docs: https://docs.astro.build/en/guides/integrations-guide/tailwind/#applybasestyles

Comment (suyash5641) [reply src/global.css:3]: @yaralahruthik  So what if we remove @tailwind base; @tailwind components; @tailwind utilities; and instead use this 

:root {
  --color-primary: 0deg 0% 0%;
  --color-secondary: 0deg 0% 24%;
  --color-hover: 0deg 0% 97%;
  --foreground: 0deg 0% 0%;
  --background: 0deg 0% 100%;
}

.dark {
  --color-hover: 0deg 0% 15%;
  --color-secondary: 255deg 100% 100%;
  --foreground: 0deg 0% 100%;
  --background: 0deg 0% 0%;
}

html {
  scrollbar-gutter: stable;
}

body {
  background-color: hsl(var(--background));
  color: hsl(var(--foreground));
}

Comment (yaralahruthik) [reply src/global.css:3]: I am not sure but I think this could cause issues. I am also not sure how Tailwind directives actually work. Have to do some reading.

> To disable this default behavior, set applyBaseStyles to false. This can be useful if you need to define your own base.css file (to include a [@layer directive](https://tailwindcss.com/docs/functions-and-directives#layer), for example). This can also be useful if you do not want base.css to be imported on every page of your project.

But just disabling that default injection and providing our own global.css seems the better idea and more aligned with how others user Tailwind.

Comment (suyash5641) [reply src/global.css:3]: I will also search again and lets see if we can find a proper solution for this .

File: tailwind.config.mjs [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
         secondary: "hsl(var(--color-secondary) / <alpha-value>)",
         hover: "hsl(var(--color-hover) / <alpha-value>)",
+        background: "hsl(var(--background))",
+        foreground: "hsl(var(--foreground))",
       },
     },
   },
+  variants: {
+    typography: ["dark"],
+  },
```
Comment (yaralahruthik) [inline tailwind.config.mjs:18]: Why is this needed?

Comment (suyash5641) [reply tailwind.config.mjs:18]: without this the prose invert won't work for dark mode , I searched and in a blog found this solution . Before using that I was not getting required result .

Comment (yaralahruthik) [reply tailwind.config.mjs:18]: Ahaa, cool!

**Review by yaralahruthik — APPROVED** [review]
Things look good now!

File: tailwind.config.mjs (hunk 4 above)
Comment (NitanJana) [reply tailwind.config.mjs:18]: This worked fine for me without the typography options. What problem were you facing without this?

Comment (suyash5641) [reply tailwind.config.mjs:18]: In dark mode the change for prose classes were not applying as expected .
